tests/src/test/java/alluxio/client/ReadOnlyMountIntegrationTest.java

```diff
@@ -0,0 +1,309 @@
+      Assert.assertEquals(e.getMessage(),
+          ExceptionMessage.MOUNT_READONLY.getMessage(dstUri, MOUNT_PATH));
+    }
+
+    dstUri = new AlluxioURI(SUB_FILE_PATH + "_renamed");
+    try {
+      mFileSystem.rename(srcUri, dstUri);
+      Assert.fail("rename should not succeed under a readonly mount.");
+    } catch (AccessControlException e) {
+      Assert.assertEquals(e.getMessage(),
+          ExceptionMessage.MOUNT_READONLY.getMessage(dstUri, MOUNT_PATH));
+    }
+  }
+
+  @Test
+  public void renameFileSrcTest() throws IOException, AlluxioException {
+    AlluxioURI srcUri = new AlluxioURI(FILE_PATH);
+    AlluxioURI dstUri = new AlluxioURI("/tmp");
+    try {
+      mFileSystem.rename(srcUri, dstUri);
+      Assert.fail("rename should not succeed under a readonly mount.");
+    } catch (AccessControlException e) {
+      Assert.assertEquals(e.getMessage(),
+          ExceptionMessage.MOUNT_READONLY.getMessage(srcUri, MOUNT_PATH));
+    }
+
+    srcUri = new AlluxioURI(SUB_FILE_PATH);
+    try {
+      mFileSystem.rename(srcUri, dstUri);
+      Assert.fail("rename should not succeed under a readonly mount.");
+    } catch (AccessControlException e) {
+      Assert.assertEquals(e.getMessage(),
+          ExceptionMessage.MOUNT_READONLY.getMessage(srcUri, MOUNT_PATH));
+    }
+  }
+
+  @Test
+  public void renameDirectoryTest() throws IOException, AlluxioException {
+    AlluxioURI srcUri = new AlluxioURI(SUB_DIR_PATH);
+    AlluxioURI dstUri = new AlluxioURI(SUB_DIR_PATH + "_renamed");
+    try {
+      mFileSystem.rename(srcUri, dstUri);
+      Assert.fail("renameDirectory should not succeed under a readonly mount.");
+    } catch (AccessControlException e) {
+      Assert.assertEquals(e.getMessage(),
+          ExceptionMessage.MOUNT_READONLY.getMessage(srcUri, MOUNT_PATH));
+    }
+  }
+
+  @Test
+  public void loadMetadataTest() throws IOException, AlluxioException {
+    AlluxioURI fileUri = new AlluxioURI(FILE_PATH);
+    mFileSystem.loadMetadata(fileUri);
+
+    fileUri = new AlluxioURI(SUB_FILE_PATH);
+    mFileSystem.loadMetadata(fileUri, LoadMetadataOptions.defaults().setRecursive(true));
+  }
+
+  @Test
+  public void openFileTest() throws IOException, AlluxioException {
+    AlluxioURI fileUri = new AlluxioURI(FILE_PATH);
+    mFileSystem.loadMetadata(fileUri);
+    Assert.assertNotNull(mFileSystem.openFile(fileUri));
+
+    fileUri = new AlluxioURI(SUB_FILE_PATH);
+    mFileSystem.loadMetadata(fileUri, LoadMetadataOptions.defaults().setRecursive(true));
+    Assert.assertNotNull(mFileSystem.openFile(fileUri));
+  }
+
+  /**
+   * Creates another directory on the local filesystem, alongside the existing Ufs, to be used as a
+   * second Ufs.
+   * @return the path of the alternate Ufs directory
+   * @throws InvalidPathException if the UNDERFS_ADDRESS is not properly formed
+   * @throws IOException if a UnderFS I/O error occurs
+   */
+  private String createAlternateUfs() throws InvalidPathException, IOException {
```

Comment:
We can't use the existing UFS path, since the mounting checks to see if a new mount point refers to a UFS path that is a subdirectory of an existing mount point. I copied this functionality from FileSystemIntegrationTest.